block-producer/src/txqueue/mod.rs

```diff
@@ -158,6 +181,8 @@ where
     TV: TransactionVerifier,
     BB: BatchBuilder,
 {
+    #[allow(clippy::blocks_in_conditions)] // Workaround of `instrument` issue
+    #[instrument(target = "miden-block-producer", skip(self, tx), err, fields(tx_id = %tx.id().inner()))]
```

Comment:
nit: instead of `skip(self, tx)` could we not say `skip(all)`?

Also, the way we are recording transaction ID in a filed is correct here. This way, it will propagate to all child spans.

One other thing: I wonder if we should also add `accountId` as a filed here as well.